repository/repository.go

```diff
@@ -406,11 +398,26 @@ func (r *Repository) Diff(ctx context.Context, id string, w io.Writer) error {
 
 	diffArgs = append(diffArgs, revisionRange)
 
-	cmd := exec.CommandContext(ctx, "git")
-	cmd.Dir = r.userRepoPath
-	cmd.Args = diffArgs
-	cmd.Stdout = w
-	cmd.Stderr = w
+	return RunInteractiveGitCommand(ctx, r.userRepoPath, w, diffArgs...)
+}
+
+func (r *Repository) Merge(ctx context.Context, id string, w io.Writer, squash bool) error {
+	envInfo, err := r.Info(ctx, id)
+	if err != nil {
+		return err
+	}
+
+	output, err := RunGitCommand(ctx, r.userRepoPath, "stash", "save", "--include-untracked", "container-use: stash before merging "+envInfo.ID)
+	if err == nil {
+		if !strings.Contains(output, "No local changes to save") {
+			defer func() {
+				_ = RunInteractiveGitCommand(ctx, r.userRepoPath, w, "stash", "pop", "-q")
+			}()
+		}
+	}
 
-	return cmd.Run()
+	if squash {
+		return RunInteractiveGitCommand(ctx, r.userRepoPath, w, "merge", "--squash", "--", "container-use/"+envInfo.ID)
```

Comment:
What happens if you pass a commit message here?

Reply:
Nothing, unfortunately